ksql-cli/src/main/java/io/confluent/ksql/cli/console/Console.java

```diff
@@ -213,58 +221,120 @@ private void registerDefaultCommands() {
-        separatorLength += columnLength + 3;
+      public Builder withRow(String... row) {
+        this.rowValues.add(Arrays.asList(row));
+        return this;
       }
 
-      String rowFormatString = constructRowFormatString(columnLengths);
+      public Builder withRow(List<String> row) {
+        this.rowValues.add(row);
+        return this;
+      }
 
-      writer().printf(rowFormatString, columnHeaders.toArray());
+      public Builder withHeaderLine(String headerLine) {
+        this.header.add(headerLine);
+        return this;
+      }
+
+      public Builder withFooterLine(String footerLine) {
+        this.footer.add(footerLine);
+        return this;
+      }
+    }
+
+    private int getMultiLineStringLength(String s) {
+      String[] split = s.split("\n");
+      String longest = Collections.max(
+          Arrays.asList(split),
+          Comparator.comparing(line -> line.length())
+      );
+      return longest.length();
+    }
 
-      writer().println(new String(new char[separatorLength]).replaceAll(".", "-"));
+    private int getColumnLength(List<String> columnHeaders, List<List<String>> rowValues, int i) {
+      int columnLength = columnHeaders.get(i).length();
       for (List<String> row : rowValues) {
-        writer().printf(rowFormatString, row.toArray());
+        columnLength = Math.max(columnLength, getMultiLineStringLength(row.get(i)));
       }
-      writer().println(new String(new char[separatorLength]).replaceAll(".", "-"));
+      return columnLength;
     }
 
-    footer.forEach(m -> writer().println(m));
+    public void print(final Console console) {
 
-    flush();
-  }
+      header.forEach(m -> console.writer().println(m));
 
-  private int getColumnLength(List<String> columnHeaders, List<List<String>> rowValues, int i) {
-    int columnLength = columnHeaders.get(i).length();
-    for (List<String> row : rowValues) {
-      columnLength = Math.max(columnLength, getMultiLineStringLength(row.get(i)));
-    }
-    return columnLength;
-  }
+      if (columnHeaders.size() > 0) {
+        console.addResult(columnHeaders, rowValues);
 
-  private int getMultiLineStringLength(String s) {
-    String[] split = s.split("\n");
-    String longest = Collections.max(
-        Arrays.asList(split),
-        Comparator.comparing(line -> line.length())
-    );
-    return longest.length();
+        Integer[] columnLengths = new Integer[columnHeaders.size()];
+        int separatorLength = -1;
+
+        for (int i = 0; i < columnLengths.length; i++) {
+          int columnLength = getColumnLength(columnHeaders, rowValues, i);
+          columnLengths[i] = columnLength;
+          separatorLength += columnLength + 3;
+        }
+
+        String rowFormatString = constructRowFormatString(columnLengths);
+
+        console.writer().printf(rowFormatString, columnHeaders.toArray());
+
+        console.writer().println(new String(new char[separatorLength]).replaceAll(".", "-"));
```

Comment:
Could use `Arrays.fill(charArray, '-')` and then create the separator string once and use here and on line 331?